### PR TITLE
Add action item for removing a passkey from entry

### LINKED ISSUE #10715: Passkeys: Remove passkey from entry
## Overview
There should be an option to remove a passkey from an entry but keep the entry with the remaining attributes.

## Steps to Reproduce
1. Register passkey to existing entry
2. Right click on the entry and look for an option to remove the passkey again

## Expected Behavior
There should be an option to remove the passkey from the selected entry.

## Actual Behavior
I couldn't find an option to remove the passkey from the selected entry.

## Context
KeePassXC - Version 2.7.8
Revision: f6757d3
Distribution: Flatpak

Operating System: Linux
Desktop Env: Gnome
Windowing System: Wayland

### PR DESCRIPTION
**Pull request by varjolintu**
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )
Adds a new menu item "Remove passkey from entry" that deletes all passkey related attributes.

Fixes #10715

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Manually.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

### CODE REVIEW

File: src/gui/DatabaseWidget.cpp [hunk 1]
```diff
@@ -1428,6 +1428,22 @@ void DatabaseWidget::showImportPasskeyDialog(bool isEntry)
...
+{
+    auto currentEntry = currentSelectedEntry();
+    if (!currentEntry) {
+        return;
+    }
+
+    auto result = MessageBox::question(this,
+                                       tr("Remove passkey from entry"),
+                                       tr("Do you want to remove the passkey from this entry?"),
+                                       MessageBox::Yes | MessageBox::No);
```
Comment (droidmonkey) [inline src/gui/DatabaseWidget.cpp]: I think this should use the "Remove" and "Cancel" buttons instead

File: src/gui/MainWindow.cpp [hunk 2]
```diff
@@ -412,6 +413,7 @@ MainWindow::MainWindow()
     m_ui->actionPasskeys->setIcon(icons()->icon("passkey"));
     m_ui->actionImportPasskey->setIcon(icons()->icon("document-import"));
     m_ui->actionEntryImportPasskey->setIcon(icons()->icon("document-import"));
+    m_ui->actionEntryRemovePasskey->setIcon(icons()->icon("document-close"));
```
Comment (droidmonkey) [inline src/gui/MainWindow.cpp:416]: Would we want to use a new icon that is the passkey icon with an X on it?

**Review by droidmonkey — CHANGES_REQUESTED** [review]

File: src/gui/MainWindow.cpp (hunk 2 above)
Comment (varjolintu) [reply src/gui/MainWindow.cpp:416]: I could try to make one.

Comment (droidmonkey) [reply src/gui/MainWindow.cpp:416]: The X is probably fine

File: src/gui/DatabaseWidget.cpp [hunk 3]
```diff
@@ -1428,6 +1428,22 @@ void DatabaseWidget::showImportPasskeyDialog(bool isEntry)
...
+    auto currentEntry = currentSelectedEntry();
+    if (!currentEntry) {
+        return;
+    }
+
+    auto result = MessageBox::question(this,
+                                       tr("Remove passkey from entry"),
+                                       tr("Do you want to remove the passkey from this entry?"),
+                                       MessageBox::Remove | MessageBox::Cancel);
+    if (result == MessageBox::Yes) {
```
Comment (droidmonkey) [inline src/gui/DatabaseWidget.cpp]: This needs to be `MessageBox::Remove`

**Review by droidmonkey — CHANGES_REQUESTED** [review]

**Review by droidmonkey — APPROVED** [review]

**Pull request merged**